lib/src/internal/http_endpoints.dart

```diff
@@ -2137,4 +2149,118 @@ class HttpEndpoints implements IHttpEndpoints {
       yield GuildEvent(rawGuildEvent as RawApiMap, client);
     }
   }
+
+  @override
+  Stream<IAutoModerationRule> fetchAutoModerationRules(Snowflake guildId) async* {
+    final response = await httpHandler.execute(
+      BasicRequest(
+        HttpRoute()
+          ..guilds(id: guildId.toString())
+          ..autoModeration()
+          ..rules(),
+      ),
+    );
+
+    if (response is IHttpResponseError) {
+      yield* Stream.error(response);
+    }
+
+    for (final rawRule in (response as IHttpResponseSuccess).jsonBody as RawApiList) {
+      final rule = AutoModerationRule(rawRule as RawApiMap, client);
+      client.guilds[guildId]?.autoModerationRules[rule.id] = rule;
+      yield rule;
+    }
+  }
+
+  @override
+  Future<IAutoModerationRule> fetchAutoModerationRule(Snowflake guildId, Snowflake ruleId) async {
+    final response = await httpHandler.execute(
+      BasicRequest(
+        HttpRoute()
+          ..guilds(id: guildId.toString())
+          ..autoModeration()
+          ..rules(id: ruleId.toString()),
+      ),
+    );
+
+    if (response is IHttpResponseError) {
+      return Future.error(response);
+    }
+
+    final rule = AutoModerationRule((response as IHttpResponseSuccess).jsonBody as RawApiMap, client);
+
+    client.guilds[guildId]?.autoModerationRules[ruleId] = rule;
+
+    return rule;
+  }
+
+  @override
+  Future<IAutoModerationRule> createAutoModerationRule(Snowflake guildId, AutoModerationRuleBuilder builder, {String? auditReason}) async {
+    final response = await httpHandler.execute(
+      BasicRequest(
+        HttpRoute()
+          ..guilds(id: guildId.toString())
+          ..autoModeration()
+          ..rules(),
+        method: 'POST',
+        auditLog: auditReason,
+        body: builder.build(),
+      ),
+    );
+
+    if (response is IHttpResponseError) {
+      return Future.error(response);
+    }
+
+    final rule = AutoModerationRule((response as IHttpResponseSuccess).jsonBody as RawApiMap, client);
+
+    client.guilds[guildId]?.autoModerationRules[rule.id] = rule;
+
+    return rule;
+  }
+
+  @override
+  Future<IAutoModerationRule> editAutoModerationRule(Snowflake guildId, Snowflake ruleId, AutoModerationRuleBuilder builder, {String? auditReason}) async {
+    final response = await httpHandler.execute(
+      BasicRequest(
+        HttpRoute()
+          ..guilds(id: guildId.toString())
+          ..autoModeration()
+          ..rules(id: ruleId.toString()),
+        body: builder.build(),
+        auditLog: auditReason,
+        method: 'PATCH',
+      ),
+    );
+
+    if (response is IHttpResponseError) {
+      return Future.error(response);
+    }
+
+    final rule = AutoModerationRule((response as IHttpResponseSuccess).jsonBody as RawApiMap, client);
+
+    client.guilds[guildId]?.autoModerationRules[ruleId] = rule;
+
+    return rule;
```

Comment:
Isn't result of assigning value to cache returning that value? If so these two lines can be shortened to:
```suggestion
    return client.guilds[guildId]?.autoModerationRules[ruleId] = rule;
```

Reply:
If `client.guilds[guildId]` returns null, then the return value of `[]=` too

Reply:
then its all right